Wbooks/Detail/Book Detail/BookDetailController.swift

```diff
@@ -0,0 +1,97 @@
+//
+//  BookDetailController.swift
+//  Wbooks
+//
+//  Created by Juan Martín Gordo on 21/04/2021.
+//
+
+import UIKit
+
+final class BookDetailController: UIViewController {
+
+    // MARK: Properties
+    private lazy var bookDetailView: BookDetailView = BookDetailView()
+    let viewModel: BookDetailViewModel
+    let errorTitle = NSLocalizedString("ALERT_BOX.TITLE", comment: "Title for the error alert box")
+    let rentErrorMessage = NSLocalizedString("ALERT_BOX.ERROR_MESSAGE", comment: "Message detailing an error in the alert box")
+    let errorMessage = NSLocalizedString("ALERT_BOX.BOOK_UNAVAILABLE_MESSAGE", comment: "Message detailing an error in the alert box")
+    let rentSuccessMessage = NSLocalizedString("ALERT_BOX.BOOK_RENTED", comment: "Message detailing an error in the alert box")
+    let errorDismiss = NSLocalizedString("ALERT_BOX.BUTTON", comment: "Text for the dismiss button on the alert box")
+
+    // MARK: Lifecycle methods
+    override func loadView() {
+        view = bookDetailView
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupView()
+        setupRentButton()
+        setupAddToWishlistButton()
+    }
+    
+    func setupView() {
+        bookDetailView.addToWishlistButton.setSecondaryStyle()
+        if viewModel.status == "Available" {
+            bookDetailView.rentButton.setMainStyle()
+        } else {
+            bookDetailView.rentButton.setUnavailableStyle()
+        }
+    }
+    
+    func setupRentButton() {
+        bookDetailView.rentButton.addTarget(self, action: #selector(rentButtonTapped), for: .touchUpInside)
+    }
+    
+    func setupAddToWishlistButton() {
+        bookDetailView.addToWishlistButton.addTarget(self, action: #selector(addToWishlistButtonTapped), for: .touchUpInside)
+    }
+
+    // MARK: Initializers
+    init(viewModel: BookDetailViewModel) {
+        self.viewModel = viewModel
+        super.init(nibName: .none, bundle: .none)
+        bookDetailView.setup(with: viewModel)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    // MARK: Actions
+    @objc func rentButtonTapped() {
+        if viewModel.status == "Available" {
+            let formatter = DateFormatter()
+            formatter.dateFormat = "yyyy-MM-dd"
+            let fromDate = Date()
+            let toDate = Calendar.current.date(byAdding: .day, value: 1, to: fromDate)
+            let params: [String : Any] = ["user_id" : "10",
+                          "book_id" : viewModel.id,
+                          "from" : formatter.string(from: fromDate),
+                          "to" : formatter.string(from: toDate!)]
+            
```

Comment:
Code related to JSON creation should be in the repository.`rentBook` should only take `onSuccess` and `onError` arguments. Then the viewModel could pass the model (book) to the repository and then this last entity would be responsable of creating the JSON.